src/components/InputField.tsx

```diff
@@ -63,17 +76,32 @@ class InputField extends React.Component<InputFiledProps, InputFieldState> {
         )
     }
 
-    // Make better names
-    onChange = (rate?: number) => {
+    changeHourlyRateState = (rate?: number) => {
         // this.setState({ [e.target.name]: e.target.value });
         this.setState({ hourlyRate: rate });
     };
+    changeInputNameState = (name?: string) => {
+        this.setState({ inputName: name });
+    };
 
-    onSubmit = () => { 
+    submitNewEmployee = () => { 
         // e.preventDefault();
-        // TODO: Sanatize input so calculation can not get corrupted
-        this.props.addSalary(200);
-        // this.props.addSalary(this.props.salaries.concat([Number(this.state.hourlyRate)]));
+
+        if(this.state.hourlyRate === undefined) return;
+        if(this.state.inputName === undefined) { this.setState({ inputName: 'Carl' }) }
+
+        const employeeList = this.props.employees
+        const incrementEmployeedId = (
+            employeeList.length > 0 
+                ? employeeList[employeeList.length-1].id+1
+                : 1
+        )
+        
+        this.props.addEmployee(employeeList.concat([{
+            id: incrementEmployeedId, 
```

Comment:
think you simplify this to 
```
id: employeeList.length // or employeeList.length+1 if you want to
```
and remove `incrementEmployeedId` (it is a complex piece of code which is hard to read, removing this kind of code is very preferable!)